tests/function-controller/framework/registry/etchosts.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2019 The Kyma Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"fmt"
+
+	"github.com/onsi/ginkgo"
+
+	appsv1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/log"
+)
+
+const (
+	etchostsManifest  = "framework/registry/manifests/sysadmin/nodes-etc-hosts-update.yaml"
+	etchostsMountPath = "/host-etc/hosts"
+
+	etchostsDaemonSetName = "registry-etc-hosts-update"
+)
+
+// patchHostEtcHosts adds an entry to the /etc/hosts file on all schedulable
+// Kubernetes nodes, and returns a function that can undo that action.
+// We do this because the DNS record of the Service is not resolvable by the
+// container runtime in most environments, yet we *must* use a domain with the
+// .local extension in order to circumvent the container runtime's requirement
+// to communicate securely (TLS) with its registry.
+func patchHostEtcHosts(f *framework.Framework, host string) (undo func()) {
```

Comment:
Addressing the registry on its IP address would actually be enough, there is no benefit to add custom entries to /etc/hosts for `.local` hostnames, but we still have an issue because TLS is enforced on the runtime (containerd) side. See https://github.com/ubuntu/microk8s/issues/384#issuecomment-502335475

Reply:
We decided not to hack the infra. We'll use a real repo instead (probably in GCE).